### ISSUE TITLE
Introduce type for domain values

### ISSUE THREAD

**Issue by ChrisJefferson** [conjure-cp/conjure-oxide#76]
This doesn't need doing now, but might be worth doing if easy, to reduce future churn.

At the moment most of conjure-oxide is using i32 for integers. Minion supports 64 bit domains (if compiled with it), and z3 (picking a random example) supports any sized integer.

Might be worth adding a `type` decleration to define domain values, which could then be tweaked later, to carefully handle the various different limits of different solvers.

**Comment by ozgurakgun** [comment]
Good idea, but let's save this for later. Could be a nice low hanging fruit for someone :)
